### PR TITLE
Fixed GraphNode port separation.

### LINKED ISSUE #32474: GraphNode port location doesn't respect Separation
**Godot version:**
3.2
**OS/device including version:**
ArcoLinux
**Issue description:**
GraphNode's ports don't seem to align correctly when Custom Constants/Separation is set to anything.  In this example, Separation is set to 40, but the same thing happens with other values:

image
(Port 2)

At first, I thought separation simply broke everything except the first port (which is positioned correctly); when attempting to drag off from a port, it would instead grab the node and move it around.  Some experimentation later, it turned out that the actual port location was just in the wrong place, and the port graphic is rendered in the right place.  There is some method to this, however: real port locations aren't where they would be without separation, but are consistently in the same, invisible spot for all but the first port.  So it seems whatever is being used to calculate that position is consistently putting it on the wrong place.

image
(Port 4)

### PR DESCRIPTION
**Pull request by levigilbert**
This fixes the spacing of the Graphnode ports when using custom separation.

"fixes #32474"

### CODE REVIEW

**Comment by KoBeWi** [comment]
You mean this issue? #32474

**Comment by lrgilbert** [comment]
Yes, that's correct.

**Comment by KoBeWi** [comment]
Why close this?

**Comment by lrgilbert** [comment]
I thought it was done?  Sorry this is my first attempt at doing a pull request, don't understand the procedure.

**Comment by KoBeWi** [comment]
The issue is still not fixed. Normally you don't manually close the PRs, they are automatically closed when they are merged. Often you have to be patient though, some PRs wait long time to be reviewed due to reasons.

**Comment by lrgilbert** [comment]
Ah ok, yeah it's been months and I thought it was dead so I was just going to remove it.

**Review by KoBeWi — APPROVED** [review]
I tested the PR and it fixes the issue without introducing any obviously visible regressions (I tried also VisualScript and VisualShader editors which also use GraphNodes AFAIK and they were fine).

I think the condition was meant to skip the separation before setting the first connection offset, so that first socket is not offseted from the top by separation. But it actually didn't make sense and resulted in skipping separation after the first socket, which caused the bug.

tl;dr the fix is ok to be merged.

**Comment by akien-mga** [comment]
Could you rebase to squash the commits into one?
https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#the-interactive-rebase

**Comment by akien-mga** [comment]
> Could you rebase to squash the commits into one?
> https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#the-interactive-rebase

Did so myself.

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.3.
